packages/dd-trace/src/proxy.js

```diff
@@ -25,7 +25,7 @@ class Tracer extends NoopProxy {
     try {
       const config = new Config(options) // TODO: support dynamic config
 
-      if (!config.isCiVisibility) {
+      if (config.remoteConfig.enabled && !config.isCiVisibility) {
```

Comment:
Hi @joeyzhao2018, @simon-id - I think we may have a larger issue than just Lambda. It seems that `remoteConfig` is fully dependent on the agent, but this code never checks for an agent to be present or configured; and in fact this library can be ran agentless, using the [logs](https://github.com/DataDog/dd-trace-js/blob/master/packages/dd-trace/src/exporters/log/index.js) exporter.

Log-based trace writing is _likely_ mainly used by Lambda, but there's nothing stopping other services (especially AWS services like Fargate, ECS, ElasticBeanstalk, AppRunner, or even EC2), from using this library and pushing APM traces to cloudwatch, which would cause this error.

It's likely only stopping Lambda (as `log.Error` creates a new instance of an error which I think the Lambda runtime is interpreting as a failure perhaps? Unclear on that exactly). But it's still probably an issue that should be addressed more widely.

Reply:
Remote config is fully dependent on the agent yes. And as the remote config RFC specifies, we need to keep requesting the agent endpoint in case an agent pops up. 
The only inconvenience this should cause is failing requests, and a debug (error level) log being generated.
What exactly is causing problems here ? if it's the log, we could potentially reduce it to a debug level.

Reply:
Hi yes Joey is right, this seems to crash the runtime in AWS Lambda's Node14.10 runtime. It's easy to reproduce (Comments mine):
```
➜  esm-demo sls run-watch --function hello
Warning: Using serverless deploy -f option only updates the function code and will not update CloudFormation stack (env variables included).
2023-01-19 10:46:28.332 INFO    ASTUYVE - enabling remote config in proxy.js
2023-01-19 10:46:28.343 INFO    ASTUYVE - BEGINNING POLL IN REMOTECONFIG MANAGER
2023-01-19 10:46:28.355 ERROR   Uncaught Exception      {"errorType":"Error","errorMessage":"connect ECONNREFUSED 127.0.0.1:8126","code":"ECONNREFUSED","errno":-111,"syscall":"connect","address":"127.0.0.1","port":8126,"stack":["Error: connect ECONNREFUSED 127.0.0.1:8126","    at TCPConnectWrap.afterConnect [as oncomplete] (net.js:1159:16)","    at TCPConnectWrap.callbackTrampoline (internal/async_hooks.js:130:17)","    at /var/runtime/deasync.js:25:9","    at _tryAwaitImport (/var/runtime/UserFunction.js:83:12)","    at _tryRequire (/var/runtime/UserFunction.js:172:5)","    at _loadUserApp (/var/runtime/UserFunction.js:219:12)","    at Object.module.exports.load (/var/runtime/UserFunction.js:300:17)","    at Object.<anonymous> (/var/runtime/index.js:43:34)","    at Module._compile (internal/modules/cjs/loader.js:1085:14)","    at Object.Module._extensions..js (internal/modules/cjs/loader.js:1114:10)"]}
2023-01-19 10:46:28.361 INFO    GOT ERROR Error: connect ECONNREFUSED 127.0.0.1:8126
    at TCPConnectWrap.afterConnect [as oncomplete] (net.js:1159:16)
    at TCPConnectWrap.callbackTrampoline (internal/async_hooks.js:130:17) {
  errno: -111,
  code: 'ECONNREFUSED',
  syscall: 'connect',
  address: '127.0.0.1',
  port: 8126
}
2023-01-19 10:46:29.011 INFO    ASTUYVE - enabling remote config in proxy.js
2023-01-19 10:46:29.027 INFO    ASTUYVE - BEGINNING POLL IN REMOTECONFIG MANAGER
2023-01-19 10:46:29.046 ERROR   Uncaught Exception      {"errorType":"Error","errorMessage":"connect ECONNREFUSED 127.0.0.1:8126","code":"ECONNREFUSED","errno":-111,"syscall":"connect","address":"127.0.0.1","port":8126,"stack":["Error: connect ECONNREFUSED 127.0.0.1:8126","    at TCPConnectWrap.afterConnect [as oncomplete] (net.js:1159:16)","    at TCPConnectWrap.callbackTrampoline (internal/async_hooks.js:130:17)","    at /var/runtime/deasync.js:25:9","    at _tryAwaitImport (/var/runtime/UserFunction.js:83:12)","    at _tryRequire (/var/runtime/UserFunction.js:172:5)","    at _loadUserApp (/var/runtime/UserFunction.js:219:12)","    at Object.module.exports.load (/var/runtime/UserFunction.js:300:17)","    at Object.<anonymous> (/var/runtime/index.js:43:34)","    at Module._compile (internal/modules/cjs/loader.js:1085:14)","    at Object.Module._extensions..js (internal/modules/cjs/loader.js:1114:10)"]}
2023-01-19 10:46:29.052 INFO    GOT ERROR Error: connect ECONNREFUSED 127.0.0.1:8126
    at TCPConnectWrap.afterConnect [as oncomplete] (net.js:1159:16)
    at TCPConnectWrap.callbackTrampoline (internal/async_hooks.js:130:17) {
  errno: -111,
  code: 'ECONNREFUSED',
  syscall: 'connect',
  address: '127.0.0.1',
  port: 8126
}
START
Unknown application error occurred
Runtime.Unknown
END Duration: 651.61 ms Memory Used: 25 MB
```